libcontainer/cgroups/fs/memory.go

```diff
@@ -34,9 +32,15 @@ func (s *MemoryGroup) Apply(d *cgroupData) (err error) {
 				return err
 			}
 		}
-		// We have to set kernel memory here, as we can't change it once
-		// processes have been attached to the cgroup.
-		if err := s.SetKernelMemory(path, d.config); err != nil {
+		// We have to limit the kernel memory here as it won't be accounted at all
```

Comment:
Check if `memory.kmem.limit_in_bytes` file is there?

On Mon, Jul 25, 2016 at 11:46 AM, Buddha Prakash <notifications@github.com>
wrote:

> In libcontainer/cgroups/fs/memory.go
> <https://github.com/opencontainers/runc/pull/962#discussion_r72121173>:
>
> > @@ -34,9 +32,15 @@ func (s *MemoryGroup) Apply(d *cgroupData) (err error) {
> >  				return err
> >  			}
> >  		}
> > -		// We have to set kernel memory here, as we can't change it once
> > -		// processes have been attached to the cgroup.
> > -		if err := s.SetKernelMemory(path, d.config); err != nil {
> > +		// We have to limit the kernel memory here as it won't be accounted at all
>
> We need to have a check here for determining if kmem is enabled on the
> system.
> @hqhq <https://github.com/hqhq> @vishh <https://github.com/vishh>
> Suggestions on how to check for the same?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencontainers/runc/pull/962/files/5106b565da1c9af39f279edb0235eb1da06d703d#r72121173>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvIKEIF-LQ6vcR1xObXWdqCQ_kmAS4aks5qZQSdgaJpZM4JTOyV>
> .
>